### PR TITLE
feat(dropdown): added support for multi-level dropdown menus

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/2635

### CODE REVIEW

**Comment by mcoker** [comment]
@mcarrano hmmm, I don't see anything different about the mockups. The shadow on the sub-menu is the same as the shadow on the main menu - the only thing that is different is its positioning.

If you'd like me to make a change so you can see what it looks like another way, let me know and I'll update the example in the PE.

**Comment by mceledonia** [comment]
Looks good to me, thanks @mcoker @mcarrano The shadow thing doesn't bother me too much, and maybe makes more sense since the expanded menu is now the "top" interest of the user as they've intently activated it.

**Comment by mattnolting** [comment]
Screen Shot 2020-02-06 at 12 29 03 PM

File: src/patternfly/components/Dropdown/examples/Dropdown.md [hunk 1]
```diff
@@ -173,15 +173,38 @@ The dropdown panel is provided for flexibility in allowing various content withi
...
+```hbs title=Multi-level
+{{#> dropdown id="dropdown-multi-level" dropdown--IsMultiLevel="true" dropdown--IsExpanded="true" dropdown--HasToggleIcon="true"}}
+  {{#> dropdown-toggle-text}}
+    Expanded dropdown
+  {{/dropdown-toggle-text}}
+{{/dropdown}}
+```
+
+```hbs title=Multi-multi-level
+{{#> dropdown id="dropdown-multi-multi-level" dropdown--IsMultiLevel="true" dropdown--IsMultiMultiLevel="true" dropdown--IsExpanded="true" dropdown--HasToggleIcon="true"}}
```
Comment (mattnolting) [inline src/patternfly/components/Dropdown/examples/Dropdown.md]: Need a unique id

File: src/patternfly/components/Dropdown/examples/Dropdown.md [hunk 2]
```diff
@@ -173,15 +173,38 @@ The dropdown panel is provided for flexibility in allowing various content withi
...
+  {{#> dropdown-toggle-text}}
+    Expanded dropdown
+  {{/dropdown-toggle-text}}
+{{/dropdown}}
+```
+
+```hbs title=Multi-level-align-right
+{{#> dropdown id="dropdown-multi-level" dropdown-menu--modifier="pf-m-align-right" dropdown--IsMultiLevel="true" dropdown--IsExpanded="true" dropdown-toggle--modifier="pf-m-plain" dropdown--HasKebabIcon="true" aria-label="Actions"}}
```
Comment (mattnolting) [inline src/patternfly/components/Dropdown/examples/Dropdown.md]: Need unique id

**Comment by mcoker** [comment]
Marking this on hold since we're going to revisit after sorting out the mobile design.

**Pull request closed without merging**